### PR TITLE
chore!: Update to latest Polars

### PR DESCRIPTION
**Pull request by stefanDeveloper**
Currently, the latest polars version is `1.7.1`. Our project https://github.com/stefanDeveloper/heiDGAF relies on this library, so it would be great to update it.

### CODE REVIEW

**Comment by jordandelbar** [comment]
Sure! I am on vacation for the next two weeks without access to my computer and I'll merge your PR once I am back. Can you aim your PR at the dev branch instead of main? I'll create a release by merging it into main afterwards. Thanks!

**Comment by stefanDeveloper** [comment]
> I realized I can change the target branch from GitHub mobile. However, the `poetry.lock` file is not up to date with the `pyproject.toml` which is causing the tests to fail. You'll need to run a `poetry lock` and push the updated `poetry.lock`

Thanks for your quick reply! I updated the `poetry.lock`.

**Comment by jordandelbar** [comment]
Thanks! I tried running the GitHub actions, but it seems I forgot to merge `main` back into `dev` after the last release. I'll need to check this on a proper terminal, not just on my phone 😅. I've added a reminder to fix this and merge your PR by the 30th of September. I also need to add a contributing guide, as the commits must follow the conventional commits specification to correctly generate the changelog.md

File: fe_polars/encoding/target_encoding.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def fit(
             # Compute the count and mean of each group
             agg = x.group_by(feature).agg(
                 [
-                    polars.count().cast(polars.Float64),
```
Comment (jordandelbar) [inline fe_polars/encoding/target_encoding.py:75]: I have opened a new pull request to update the polars version [here](https://github.com/jordandelbar/feature-engineering-polars/pull/38) in order to have the commit messages that respect the conventional commit spec (to correctly determine the version number change and generate the changelog). Can you explain why you change this line? it seems that the tests pass in 1.8.2 with the previous code (I thought you changed it due to a API change in polars from 0.19 -> 1.x.x). Thanks!

Comment (stefanDeveloper) [reply fe_polars/encoding/target_encoding.py:75]: As referred to the Polars documentation: https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.count.html, `count()` is deprecated and should be replaced with `len()`

Comment (jordandelbar) [reply fe_polars/encoding/target_encoding.py:75]: My bad, you are right! I skimmed through the doc without seeing the deprecation warning. Thank you!

**Comment by jordandelbar** [comment]
Changes were merged in https://github.com/jordandelbar/feature-engineering-polars/pull/38
Thanks again for the help!

**Pull request closed without merging**